core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,288 @@
+    await this.page.click(saveOutcomeDestButton);
+  }
+
+  /**
+   * Function to save an exploration draft.
+   */
+  async saveExplorationDraft(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      const element = await this.page.$(mobileNavbarOptions);
+      if (!element) {
+        await this.clickOn(mobileOptionsButton);
+      }
+      await this.clickOn(mobileSaveChangesButton);
+      await this.clickOn(saveDraftButton);
+      await this.page.waitForNetworkIdle();
+    } else {
+      await this.clickOn(saveChangesButton);
+      await this.clickOn(saveDraftButton);
+      await this.page.waitForNetworkIdle();
+    }
+  }
+
+  /**
+   * Function to navigate to a specific card in the exploration.
+   * @param {string} cardName - The name of the card to navigate to.
+   */
+  async navigateToCard(cardName: string): Promise<void> {
+    if (cardName === 'Introduction' && this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileStateGraphResizeButton);
+      await this.page.waitForSelector(mobileStateNodeSelector);
+      const elements = await this.page.$$(mobileStateNodeSelector);
+      await elements[3].click();
+    } else if (cardName === 'Introduction') {
+      await this.page.waitForSelector(stateNodeSelector);
+      const elements = await this.page.$$(stateNodeSelector);
+      await elements[0].click();
+    } else if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileStateGraphResizeButton);
+      await this.page.waitForSelector(mobileStateNodeSelector);
+      const elements = await this.page.$$(mobileStateNodeSelector);
+      if (cardName === 'Test Question ') {
+        await elements[3].click();
+      } else {
+        await elements[5].click();
+      }
+      await this.page.waitForNetworkIdle({idleTime: 700});
+    } else {
+      await this.clickOn(cardName);
+      await this.page.waitForNetworkIdle({idleTime: 700});
+    }
+  }
+
+  /**
+   * Function to add responses to the interactions.
+   * @param {string} response - response to be added.
+   */
+  async addResponseToTheInteraction(response: string): Promise<void> {
+    await this.type(floatFormInput, response);
+    await this.clickOn(feedbackEditorSelector);
+    await this.type(stateContentInputField, 'Correct Answer, You got that!');
+    await this.clickOn(correctAnswerInTheGroupSelector);
+    await this.clickOn(addNewResponseButton);
+    await this.page.waitForSelector(resonseModalHeaderSelector, {hidden: true});
+  }
+
+  /**
+   * Function to navigate to the preview tab.
+   */
+  async navigateToPreviewTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileNavbarDropdown);
+      await this.page.waitForSelector(mobileNavbarPane);
+      await this.clickOn(mobilePreviewTabButton);
+      await this.page.waitForNavigation();
+    } else {
+      await this.clickOn(previewTabButton);
+      await this.page.waitForNavigation();
```

Comment:
Same note about moving common stuff outside the if/else

Reply:
Done.